### PR TITLE
feat: Move ENABLE_CONFIRMATION_REDESIGN feature flag to the developer…

### PR DESCRIPTION
**Pull request by pnarayanaswamy**
… settings page #25520

## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26095?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/confirmations/signatures/permit.spec.ts [hunk 1]
```diff
@@ -14,12 +14,8 @@ import {
...
 import { Driver } from '../../../webdriver/driver';
 
-describe('Confirmation Signature - Permit', function (this: Suite) {
-  if (!process.env.ENABLE_CONFIRMATION_REDESIGN) {
-    return;
-  }
-
-  it('initiates and confirms', async function () {
+describe('Confirmation Signature - Permit @no-mmi', function (this: Suite) {
+  it('initiates and confirms and emits the correct events', async function () {
```
Comment (digiwand) [inline test/e2e/tests/confirmations/signatures/permit.spec.ts:18]: +1 extra line to resolve merge conflict - LGTM  ([compare line](https://github.com/MetaMask/metamask-extension/pull/25520/files#diff-9b7dc3b211ccc46387c285548ff2177a7f4f499a74715bba074069f0c7f82222L26))

File: test/e2e/tests/confirmations/transactions/contract-interaction-redesign.spec.ts [hunk 2]
```diff
@@ -212,7 +226,10 @@ describe('Confirmation Redesign Contract Interaction Component', function () {
           fixtures: new FixtureBuilder({ inputChainId: CHAIN_IDS.OPTIMISM })
             .withPermissionControllerConnectedToTestDapp()
             .withPreferencesController({
-              preferences: { redesignedConfirmationsEnabled: true },
+              preferences: {
+                redesignedConfirmationsEnabled: true,
+                isRedesignedConfirmationsDeveloperEnabled: true,
+              },
```
Comment (digiwand) [inline test/e2e/tests/confirmations/transactions/contract-interaction-redesign.spec.ts:232]: this file does not need to set an extra set of preferences that the original has. The original changes 4 more lines due to differences in the tests between Version v12.2.0 and develop ([compare](https://github.com/MetaMask/metamask-extension/pull/25520/files#diff-9b7dc3b211ccc46387c285548ff2177a7f4f499a74715bba074069f0c7f82222L26))

+1 (line change in above comment) - 4 not added here = -3 added lines changed. This is how we have `+469 -266` diff here and `+472 -266` in the original PR

**Review by digiwand — APPROVED** [review]
Lgtm! 

there are some tests passing that may be out of scope of this cherry-pick PR

File: test/e2e/tests/confirmations/transactions/contract-interaction-redesign.spec.ts (hunk 2 above)
Comment (digiwand) [reply test/e2e/tests/confirmations/transactions/contract-interaction-redesign.spec.ts:232]: explaining how the line diff makes sense here 👍🏼  LGTM!

**Review by pedronfigueiredo — APPROVED** [review]

**Review by danjm — APPROVED** [review]

**Pull request merged**